techniques/html/H-creating-dialogs-with-the-html-dialog-element.html

```diff
@@ -0,0 +1,215 @@
+         
+<pre xml:space="preserve"><code class="language-css">*, *::after, *::before {
+  box-sizing: inherit;
+}
+
+body {
+  background:#fff;
+  color:#000;
+  font:1rem/1.5 system-ui, Helvetica, Roboto, sans-serif;
+}
+
+*:focus-visible{
+  outline:1px solid #0054AE;
+  outline-offset:1px;
+}
+
+dialog {
+  border:1px solid #000;
+  padding:2rem;
+  position:relative;
+}
+ 
+dialog::backdrop {
+  background-color:hsla(0, 0%, 0%, .5);
+}
+ 
+.close-modal {
+  inset-block-start:1.5rem;  
+  inset-inline-end:1.5rem;  
+  line-height:1.3;
+  padding:0.25em 0.5em;
+  position:absolute;
+}
+
+.sign-up{
+  background:#000;
+  color:#fff;
+  padding:0.25em;
+}
+
+dialog h1 {
+  display:inline-block;
+  line-height:1.3333;
+  margin:0;
+  max-inline-size:95%;
+}
+
+form{
+  display:grid;
+  grid-gap:20px;
+  grid-template-columns:repeat(auto-fit, minmax(150px, 1fr));
+}
+
+.req-note, .sign-up{
+  grid-column:1 / -1;
+}
+ 
+label {
+  display:block;
+}
+ 
+input {
+  border:1px solid hsl(0, 0%, 50%);
+  font:inherit;
+  inline-size:calc(100% - 4px);
+}
+
+button{
+  background:#fff;
+  border:1px solid hsl(0, 0%, 50%);
+  border-radius:3px;
+  color:inherit;
+  font:inherit;
+  margin:0;
+}</code></pre>
+     
+		<h4>The JavaScript</h4>
+		<p>The script toggles the display of the <code class="language-html">dialog</code>. The HTML <code class="language-html">dialog</code> element can be opened using two different commands: <code class="language-javascript">show()</code> (for non-modal <code class="language-html">dialog</code>s), and  <code class="language-javascript">showModal()</code> (for modal <code class="language-html">dialog</code>s).</p>
+         
+<pre xml:space="preserve"><code class="language-javascript">document.addEventListener("DOMContentLoaded", function(e){
+  const d = document.querySelector("dialog");
+  const btnOpen = document.querySelector(".open-modal");
+  const btnClose = document.querySelector(".close-modal");
+
+  btnOpen.addEventListener("click", function(){
+    d.showModal();
+  }, false);
+
+  btnClose.addEventListener("click", function(){
+    d.close();
+  }, false);
+});</code></pre>
+         
+      </section>
+   </section><section id="tests"><h2>Tests</h2>
+      <section class="procedure"><h3>Procedure</h3>
+         <ol>
+            <li>Find the components on the page that trigger modal <code class="language-html">dialog</code> elements.</li>
+            <li>Check that each <code class="language-html">dialog</code> can be opened using the keyboard to activate the triggering element (for example: activate a <code class="language-html">button</code> by pressing the <kbd>Enter</kbd> key or <kbd>Space Bar</kbd>.</li>
+            <li>When the <code class="language-html">dialog</code> is opened, check that focus is placed inside it.</li>
+            <li>While the <code class="language-html">dialog</code> is open, check that focus is trapped inside the <code class="language-html">dialog</code>.</li>
```

Comment:
this should be changed to rather state that focus does not return to elements of the primary document.  Focus is not entirely trapped in an HTML dialog element, after all, as focus can leave and access the browser chrome.  So, this wording might confuse people into think this expected behavior is a bug.

Reply:
Thanks for that feedback. I thought I had that in there to start with, but apparently I hadn't. I've updated the text.

Reply:
looks like this change wasn't made.  
```suggestion
            <li>While the <code class="language-html">dialog</code> is open, check that keyboard focus cannot move to elements of the primary document.</li>
```